libcxx/test/tools/clang_tidy_checks/CMakeLists.txt

```diff
@@ -37,7 +37,7 @@ set(Clang_DIR "${Clang_DIR_SAVE}" CACHE PATH "The directory containing a CMake c
 
 message(STATUS "Found system-installed LLVM ${LLVM_PACKAGE_VERSION} with headers in ${LLVM_INCLUDE_DIRS}")
 
-set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD 23)
```

Comment:
We shouldn't update these requirements for now. This tool uses the system compiler and system standard library. IMO these requirement should be the general LLVM requirements. Maybe we should add some comments to that effect.

Reply:
Am I missing something? AFAIK this uses the same compiler as the rest of libc++, which will almost certainly also entail an up-to-date stdandard library.

Reply:
It does not. I my system (Debian) it uses the system compiler (gcc+) and libstdc++.

Reply:
For me it fails even after using a clean CMake build. How do you configure CMake using `CXX=foo cmake ...` or `cmake -DCMAKE_CXX_COMPILER=foo`? I do the latter.

Reply:
I set environment variables. Could you try that version on your system? If that works, there is something weird going on. Also, how are you getting your clang package? Maybe this is similar to how `LLVM_DIR` and `Clang_DIR` change for some reason?

Reply:
Using environment variables has no effect. 
Looking at the compilation command for the clang-tidy plugin the command used is
```
ccache /usr/lib/llvm-18/bin/clang++ -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS
-D__STDC_LIMIT_MACROS -Dcxx_tidy_EXPORTS -isystem /usr/lib/llvm-17/include -fvisibility-inlines-hidden
-Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings
-Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type
-Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation
-Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG -std=c++20
-fPIC -fno-rtti -fno-sanitize=address,hwaddress,undefined,thread,leak -fno-sanitize=memory,dataflow
-MD -MT libcxx/test/tools/clang_tidy_checks/CMakeFiles/cxx-tidy.dir/header_exportable_declarations.cpp.o
-MF libcxx/test/tools/clang_tidy_checks/CMakeFiles/cxx-tidy.dir/header_exportable_declarations.cpp.o.d
-o libcxx/test/tools/clang_tidy_checks/CMakeFiles/cxx-tidy.dir/header_exportable_declarations.cpp.o
-c /data/src/llvm/libcxx/test/tools/clang_tidy_checks/header_exportable_declarations.cpp
```
I see nowhere that it tries to use the built libc++, instead it uses the system library.

Reply:
I would expect this to use the compiler chosen with `CMAKE_CXX_COMPILER` and its associated library as well. Which on Linux is going to be libstdc++ usually, regardless of whether `CMAKE_CXX_COMPILER` is Clang or GCC. I think that's fine. It's a bit concerning that GCC 13 doesn't seem to accept libstdc++'s `<ranges>` header when `-std=c++23` is passed but I guess that's separate. I'll revert this part of the patch simply because it will help unblock the patch by fixing some of the CI, and concretely it's not critical to be building the Clang tidy checks with the latest Standard mode.